### PR TITLE
Add repo administration commands to googet

### PR DESCRIPTION
**Pull request by adjackura**
Add addrepo, rmrepo, and listrepos commands

### CODE REVIEW

File: googet_rmrepo.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+	repoEntries, err := repos(filepath.Join(rootDir, repoDir))
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	var repoPath string
+	for _, re := range repoEntries {
+		if strings.ToLower(re.Name) == strings.ToLower(name) {
+			repoPath = re.fileName
```
Comment (jrunningen) [inline googet_rmrepo.go]: Is it possible for more than one repoFile to have the same name? If so, this command will only remove the last one it finds.

File: googet_rmrepo.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	rfs, err := unmarshalRepoFile(repoPath)
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	for i, rf := range rfs {
+		if strings.ToLower(rf.Name) == strings.ToLower(name) {
+			rfs = append(rfs[:i], rfs[i+1:]...)
```
Comment (jrunningen) [inline googet_rmrepo.go]: The way you're modifying the slice as you iterate over it isn't safe. See https://play.golang.org/p/PAMc9s-a4i.

File: googet.go [hunk 3]
```diff
@@ -98,11 +107,18 @@ func unmarshalRepoFile(p string) ([]repoFile, error) {
...
 		return []repoFile{rf}, nil
 	}
 
 	var rfs []repoFile
-	return rfs, yaml.Unmarshal(b, &rfs)
+	if err := yaml.Unmarshal(b, &rfs); err != nil {
+		return nil, err
+	}
+	for i := range rfs {
+		rfs[i].fileName = p
```
Comment (jrunningen) [inline googet.go]: Assigning fileName to the same thing for every element in rfs seems odd. Can you clarify why this is necessary?

File: googet_addrepo.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	rfs, err := unmarshalRepoFile(repoPath)
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	for i, rf := range rfs {
+		if rf.Name == name || rf.URL == url {
+			rfs = append(rfs[:i], rfs[i+1:]...)
```
Comment (jrunningen) [inline googet_addrepo.go]: Unsafe slice modification. See https://play.golang.org/p/PAMc9s-a4i.

File: googet_rmrepo.go (hunk 1 above)
Comment (adjackura) [inline googet_rmrepo.go]: No, as they are all in the same directory only one per name

File: googet_rmrepo.go (hunk 2 above)
Comment (adjackura) [inline googet_rmrepo.go]: could've sworn i had this fixed, must not have submitted it

File: googet_addrepo.go (hunk 4 above)
Comment (adjackura) [inline googet_addrepo.go]: done

File: googet.go (hunk 3 above)
Comment (adjackura) [inline googet.go]: Each repoFile element has a field for where the repo file it belongs to is located. This was the simplest way to record this info without changing upstream functions too much.

File: googet_rmrepo.go (hunk 1 above)
Comment (adjackura) [inline googet_rmrepo.go]: ah i see what your saying, and yes that would be possible but I chose not to catch it with these functions

File: system/system_windows.go [hunk 5]
```diff
@@ -63,11 +64,80 @@ func addUninstallEntry(dir string, ps *goolib.PkgSpec) error {
...
+	}
+	for key, val := range envMap {
+		if err := os.Setenv(key, val); err != nil {
+			logger.Errorf("Error setting environment: %v", err)
+		}
+	}
+	for key, val := range exEnvMap {
+		exVal, err := registry.ExpandString(val)
+		if err != nil {
+			logger.Error(err)
```
Comment (jrunningen) [inline system/system_windows.go]: Continue here?

File: system/system_windows.go [hunk 6]
```diff
@@ -63,11 +64,80 @@ func addUninstallEntry(dir string, ps *goolib.PkgSpec) error {
 }
 
 func removeUninstallEntry(name string) error {
-	reg := regBase + "GooGet - " + name
+	reg := uninstallBase + "GooGet - " + name
 	logger.Infof("Removing uninstall entry %q from registry.", reg)
 	return registry.DeleteKey(registry.LOCAL_MACHINE, reg)
 }
 
+func refreshEnv() {
```
Comment (jrunningen) [inline system/system_windows.go]: This function is big enough to justify a comment explaining what it does.

**Comment by jrunningen** [comment]
LGTM

**Pull request merged**